### PR TITLE
improve-code-and-add-gs-ci

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR:

- adds giantswarm CI tooling
- makes proxy config types public domain so they can in used in atlas operators

This makes the https://github.com/giantswarm/grafana-multi-tenant-proxy/pull/152 easier to review

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,17 +1,24 @@
----
 version: 2.1
 
 orbs:
-  architect: giantswarm/architect@5.6.0
+  architect: giantswarm/architect@dev:30a70d1ee2cf158cd0ea66d74cf4b0bf930009d3
```
Comment (hervenicol) [inline .circleci/config.yml:4]: won't renovate try to revert this ASAP?

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,7 +9,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
-- [breaking] Refactor configuration so that `target server` or `keep-org-id` are now configured via config instead of via a cli flag.
+- Move config package from `internal` to public packages so it can be imported in our operators.
+- Use giantswarm CI tooling to build and improve error handling.
```
Comment (hervenicol) [inline CHANGELOG.md:13]: I think these are 2 unrelated changes, right?
```suggestion
- Use giantswarm CI tooling to build
- improve error handling.
```

File: .circleci/config.yml (hunk 1 above)
Comment (QuentinBisson) [reply .circleci/config.yml:4]: Yes probably. Honestly, i'm waiting for @TheoBrigitte PR to be merged

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,17 +1,25 @@
----
 version: 2.1
 
 orbs:
   architect: giantswarm/architect@5.7.0
 
+
```
Comment (QuentinBisson) [inline .circleci/config.yml:6]:
```suggestion
```

**Review by hervenicol — APPROVED** [review]
Fine with me :+1:
